core/platforms/packaging/bareos.spec

```diff
@@ -524,11 +526,37 @@ Requires:   %{name}-common = %{version}
 Summary:    Python plugin for Bareos Director daemon
 Group:      Productivity/Archiving/Backup
 Requires:   bareos-director = %{version}
+Requires:   bareos-director-python-plugins-common = %{version}
+
+%package    director-python3-plugin
+Summary:    Python plugin for Bareos Director daemon
+Group:      Productivity/Archiving/Backup
+Requires:   bareos-director = %{version}
+Requires:   bareos-director-python-plugins-common = %{version}
```

Comment:
If you want the python-based-plugins to depend on the python-c-plugin, you will need a common virtual capability provided by both `*-python-plugin` and `*-python3-plugin` packages each and require that capability in the plugin packages.

However, historically `yum`'s depsolver (I'm not sure if `dnf` is the same) always selects the package with the shortest name when selecting the package to fulfil a dependency. This had been introduced to make sure we do not install a `*-debuginfo` package that would fulfill the dependency.
With the current naming scheme (i.e. `python` vs. `python3` you will most likely end up with the python2 plugin installed (at least on RHEL 7).